### PR TITLE
Multi_precision support for Faster R-CNN and Mask R-CNN

### PR DESCRIPTION
**Pull request by karan6181**
- Improved the throughput of approx. 5% on multi-node without sacrificing the accuracy.

cc @Jerryzcn

### CODE REVIEW

**Comment by mli** [comment]
Job PR-1099-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1099/1/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1099-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1099/2/index.html 
Code coverage of this PR:  vs. Master:

File: scripts/instance/mask_rcnn/eval_mask_rcnn.py [hunk 1]
```diff
@@ -113,8 +115,7 @@ def validate(net, val_data, ctx, eval_metric, size):
                     im_height, im_width = int(round(im_height / im_scale)), int(
                         round(im_width / im_scale))
                     full_masks = gcv.data.transforms.mask.fill(det_mask, det_bbox,
-                                                               (im_width, im_height),
-                                                               fast_fill=False)
+                                                               (im_width, im_height))
```
Comment (Jerryzcn) [inline scripts/instance/mask_rcnn/eval_mask_rcnn.py]: Can u set fast_fill=False

File: scripts/detection/faster_rcnn/train_faster_rcnn.py [hunk 2]
```diff
@@ -114,6 +114,8 @@ def parse_args():
     parser.add_argument('--kv-store', type=str, default='nccl',
                         help='KV store options. local, device, nccl, dist_sync, dist_device_sync, '
                              'dist_async are available.')
+    parser.add_argument('--dtype', type=str, default='float32',
```
Comment (Jerryzcn) [inline scripts/detection/faster_rcnn/train_faster_rcnn.py]: Can u add some documentation about the difference between AMP and float16? or we could merge float16 and AMP into one option?

File: scripts/instance/mask_rcnn/eval_mask_rcnn.py (hunk 1 above)
Comment (karan6181) [reply scripts/instance/mask_rcnn/eval_mask_rcnn.py]: Done [fffe39a](https://github.com/dmlc/gluon-cv/pull/1099/commits/fffe39a15920a22e941ede962809b56a75f1ee19). Also fixed a bug which surfaces by setting `fast_fill=False`.

**Comment by mli** [comment]
Job PR-1099-3 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1099/3/index.html 
Code coverage of this PR:  vs. Master:

File: scripts/detection/faster_rcnn/train_faster_rcnn.py (hunk 2 above)
Comment (karan6181) [reply scripts/detection/faster_rcnn/train_faster_rcnn.py]: Merged AMP and float16 into one option and also added a small comments about float16.

**Comment by mli** [comment]
Job PR-1099-4 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1099/4/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**